table/meta_blocks.cc

```diff
@@ -347,32 +349,46 @@ Status ReadTablePropertiesHelper(
         continue;
       }
       *(pos->second) = val;
+      new_table_properties->IncreaseApproximateMemoryUsage(sizeof(val));
     } else if (key == TablePropertiesNames::kDbId) {
       new_table_properties->db_id = raw_val.ToString();
+      new_table_properties->IncreaseApproximateMemoryUsage(raw_val.size());
```

Comment:
Not my favorite code repetition either but couldn't find an easy way to avoid that ...

Reply:
Will fix this (previously I avoided it due to the nested if-else but now I think it is still better than code duplication)

Reply:
I think one "longer-term" fix is to avoid "key ==  TablePropertiesNames::XX" pattern and use set. Opened a tech debt for it.

Reply:
Done by on-the-fly memory approximation for table properties